### PR TITLE
BQ e2e updated tests

### PR DESCRIPTION
**Pull request by priyabhatnagar25**

### CODE REVIEW

File: src/e2e-test/features/bigquery/source/BigQueryToBigQuery_withConnections.feature
Comment (Aryan-Verma) [inline src/e2e-test/features/bigquery/source/BigQueryToBigQuery_withConnections.feature]: Why are we removing this?

File: src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java]: nit:
```suggestion
  public static WebElement bqmtAllowFlexibleSchemasInOutput;
```

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: we can change the name to compareSourceDataWithTargetData

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java:16]: Please add license

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuery.java
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuery.java]:
```suggestion
 * Copyright © 2023 Cask Data, Inc.
```

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (Aryan-Verma) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java]:
```suggestion
 * Copyright © 2023 Cask Data, Inc.
```

File: src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java
Comment (sahusanket) [inline src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java]: All these locators are already present in the framework repo: https://github.com/cdapio/cdap-e2e-tests/blob/develop/src/main/java/io/cdap/e2e/pages/locators/CdfBigQueryPropertiesLocators.java

Why are we re-inventing the wheel here?

**Review by sahusanket — CHANGES_REQUESTED** [review]

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (sahusanket) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java:2]: I think it should be `2021-2023`

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (sahusanket) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java]: These test setup hooks are already present here: https://github.com/data-integrations/google-cloud/blob/develop/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

Why are we adding them again here?

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java
Comment (sahusanket) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: Why are these removed?

File: src/e2e-test/features/bigquery/source/BigQueryToBigQuery_withConnections.feature
Comment (bharatgulati) [reply src/e2e-test/features/bigquery/source/BigQueryToBigQuery_withConnections.feature]: Reverted

File: src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java]: Done.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: Done.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java:16]: Done

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuery.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQuery.java]: Done

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java]: Done

File: src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java]: Reverting the changes

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java]: Reverting the changes

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBase.java]: Added back.

**Review by Aryan-Verma — APPROVED** [review]

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/TestSetupHooks.java]: Reverted. @sahusanket

File: src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/locators/BigQueryLocators.java]: Reverted.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature
Comment (itsankit-google) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Replace this step with a generic step definiton like: https://github.com/cdapio/cdap-e2e-tests/blob/4ec16f7af29ffcfb5e44cf549ec7a919cf44d2fa/src/main/java/stepsdesign/PipelineSteps.java#L457

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature
Comment (itsankit-google) [inline src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: similarly here, replace it with a generic step.

File: src/e2e-test/java/io/cdap/plugin/bigquery/actions/BigQueryActions.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/bigquery/actions/BigQueryActions.java]: remove this class, similar class is present in `cdap-e2e-tests` repo, any changes should be maintained at one place.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature
Comment (bharatgulati) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Adding a generic step according to these fields (tableKey, deDupeBy and clusteringOrder) as a locator is different from the one which we used in Cdap e2e framework step.

Comment (itsankit-google) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: I know it is different, my comment was to add a similar generic locator in cdap-e2e-framework and use it here.

Comment (bharatgulati) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Added.

File: src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature
Comment (bharatgulati) [reply src/e2e-test/features/bigquery/sink/BigQueryToBigQuerySink.feature]: Replaced with a generic step.

File: src/e2e-test/java/io/cdap/plugin/bigquery/actions/BigQueryActions.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/actions/BigQueryActions.java]: Removed.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: why this change? Looks like nothing is changed in this file.

Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: An unintended step was initially removed and is now added. Aryan has also suggested changing the year to 2023 as part of the same change.

Comment (itsankit-google) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: you said some step definition was added, but I still don't anything changed in this file, am I missing something?

**Review by itsankit-google — CHANGES_REQUESTED** [review]
Build is failing with some duplicate step definitons, please fix:

```
Duplicate step definitions in io.cdap.plugin.bigquery.stepsdesign.BigQuery.validateTheValuesOfRecordsTransferredToGcsBucketIsEqualToTheValuesFromSourceBigQueryTable() in file:/tmp/google-cloud/google-cloud/plugin/target/test-classes/ and io.cdap.plugin.gcs.stepsdesign.GCSBase.validateTheValuesOfRecordsTransferredToGcsBucketIsEqualToTheValuesFromSourceBigQueryTable() in file:/tmp/google-cloud/google-cloud/plugin/target/test-classes/
```

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryBase.java]: I Apologize for the misunderstanding. The changes were made to the GCS Base, not the BigQueryBase. Reverted the year back to 2021. [commit link](https://github.com/data-integrations/google-cloud/pull/1273/commits/7730c6e09d09620852af5908b7827d9d56a416e3)

**Comment by bharatgulati** [comment]
> Build is failing with some duplicate step definitons, please fix:
> 
> ```
> Duplicate step definitions in io.cdap.plugin.bigquery.stepsdesign.BigQuery.validateTheValuesOfRecordsTransferredToGcsBucketIsEqualToTheValuesFromSourceBigQueryTable() in file:/tmp/google-cloud/google-cloud/plugin/target/test-classes/ and io.cdap.plugin.gcs.stepsdesign.GCSBase.validateTheValuesOfRecordsTransferredToGcsBucketIsEqualToTheValuesFromSourceBigQueryTable() in file:/tmp/google-cloud/google-cloud/plugin/target/test-classes/
> ```

Resolved.

**Comment by itsankit-google** [comment]
> Resolved.

Bigquery tests are still failing and does not look like related to existing chrome driver issue, ptal, thanks!

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: remove extra empty lines.

Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: Removed.

File: src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: what are these hardcoded UUIDs??

Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BQValidation.java]: The main function for debugging and fixing e2e build failures with source and sink tables was added to check on local. Now that they've passed. I've now removed it.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: The value of dataset should come from plugin properties.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: similar comment at all places using the hard-coded value.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Using a dataset constant instead of a hard-coded value.

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: Resolved.

**Review by itsankit-google — APPROVED** [review]
Plz squash commits before merge.

**Review by sahusanket — APPROVED** [review]

**Comment by bharatgulati** [comment]
All commits have been squashed. However, before you merge the PR. We will raise an another PR for the review of the BQ step in the cdap-e2e-framework because three steps have been skipped

**Comment by itsankit-google** [comment]
> All commits have been squashed. However, before you merge the PR. We will raise an another PR for the review of the BQ step in the cdap-e2e-framework because three steps have been skipped

Sure, will wait for the framework PR to be merged.

**Comment by bharatgulati** [comment]
> > All commits have been squashed. However, before you merge the PR. We will raise an another PR for the review of the BQ step in the cdap-e2e-framework because three steps have been skipped
> 
> Sure, will wait for the framework PR to be merged.

Please review the step added for Big Query. [commit Link](https://github.com/cdapio/cdap-e2e-tests/pull/194)

**Pull request merged**
